stan/math/rev/core/vari.hpp

```diff
@@ -180,29 +180,41 @@ using vari = vari_value<double>;
 template <typename T, typename = void>
 class vari_view;
 
+/**
+ * This struct is follows the CRTP for methods common to `vari_view<>` and
+ * `vari_value<Matrix>`.
+ */
 template <typename T>
-class vari_view<T, require_not_plain_type_t<T>> final : public vari_base {
- public:
-  using PlainObject = plain_type_t<T>;
-  using value_type = std::decay_t<T>;  // The underlying type for this class
+struct vari_view_crtp {
   /**
-   * Number of rows known at compile time
+   * Helper function to return a reference to the derived type
    */
-  static constexpr int RowsAtCompileTime = PlainObject::RowsAtCompileTime;
+  inline T& derived() { return static_cast<T&>(*this); }
   /**
-   * Number of columns known at compile time
+   * Helper function to return a constant reference to the derived type
    */
-  static constexpr int ColsAtCompileTime = PlainObject::ColsAtCompileTime;
+  inline T const& derived() const { return static_cast<T const&>(*this); }
 
-  T val_;
-  T adj_;
-  template <typename S, typename K,
-            require_convertible_t<S&, value_type>* = nullptr,
-            require_convertible_t<K&, value_type>* = nullptr>
-  vari_view(S&& val, K&& adj) noexcept : val_(val), adj_(adj) {}
+  /**
+   * Helper function to return a reference to the derived type's value.
+   */
+  inline auto& derived_val() { return static_cast<T&>(*this).val_; }
+  /**
+   * Helper function to return a constant reference to the derived type's value.
+   */
+  inline auto const& derived_val() const {
+    return static_cast<T const&>(*this).val_;
+  }
+
+  /**
+   * Helper function to return a reference to the derived type's adjoint.
+   */
+  inline auto& derived_adj() { return static_cast<T&>(*this).adj_; }
+  /**
+   * Helper function to return a reference to the derived type's adjoint.
+   */
+  inline auto& derived_adj() const { return static_cast<T&>(*this).adj_; }
```

Comment:
I don't think we need these as separate functions. We can always use `derived().val_`.

Reply:
Yeah fair